nautobot/ipam/tests/test_models.py

```diff
@@ -295,6 +296,43 @@ def setUp(self):
         self.child1 = Prefix.objects.create(prefix="101.102.0.0/26", status=self.status, namespace=self.namespace)
         self.child2 = Prefix.objects.create(prefix="101.102.0.64/26", status=self.status, namespace=self.namespace)
 
+    def test_location_queries(self):
+        PrefixFactory.create_batch(10)
+        location = Prefix.objects.filter(locations__isnull=False).first().locations.first()
+
+        with self.subTest("Assert filtering and excluding `location`"):
+            self.assertQuerysetEqualAndNotEmpty(
+                Prefix.objects.filter(location=location),
+                Prefix.objects.filter(locations__in=[location]),
+            )
+            self.assertQuerysetEqualAndNotEmpty(
+                Prefix.objects.exclude(location=location),
+                Prefix.objects.exclude(locations__in=[location]),
+            )
+            self.assertQuerysetEqualAndNotEmpty(
+                Prefix.objects.filter(location__in=[location]),
+                Prefix.objects.filter(locations__in=[location]),
+            )
+            self.assertQuerysetEqualAndNotEmpty(
+                Prefix.objects.exclude(location__in=[location]),
+                Prefix.objects.exclude(locations__in=[location]),
+            )
+
+        # These fields cannot be null, Hence using these for the test
```

Comment:
Do we have a problem with nullable fields that we need to document or fix?

Reply:
`assertQuerysetEqualAndNotEmpty` is used for test validation. Including a nullable field could lead to flaky tests where querysets might return `None`, causing tests to fail.